components/form/FFileUpload.vue

```diff
@@ -0,0 +1,259 @@
+   */
+  allowedMimeTypes?: string[];
+  /**
+   * Maximum allowed file size, in kilobytes (Kb)
+   */
+  maximumSize?: number;
+  /**
+   * Rules form validation
+   */
+  rules?: ValidationRule | ValidationRule[];
+  /**
+   * Icon displayed inside the button
+   */
+  iconName?: Icon;
+  /**
+   * Button size
+   */
+  size?: FButtonSize;
+  /**
+   * Button text
+   */
+  buttonText?: string;
+  /**
+   * Button background color
+   */
+  color?: string;
+  /**
+   * Button text color
+   */
+  textColor?: string;
+  /**
+   * Label, placed on top of input
+   */
+  label?: string;
+  /**
+   * Disable the field
+   */
+  disabled?: boolean;
+  /**
+   * Text color of the label
+   */
+  labelTextColor?: Color;
+  /**
+   * Text and caret error color
+   */
+  errorColor?: Color;
+  /**
+   * Text color of the hint
+   */
+  hintTextColor?: Color;
+  /**
+   * Hide or not the hint / error message
+   */
+  hideHint?: boolean;
+  /**
+   * A hint to display under the input
+   */
+  hint?: string;
+  /**
+   * Icon, displayed before the hint
+   */
+  hintIcon?: Icon | null;
+  /**
+   * Field name. Used in a form context
+   */
+  name?: string;
+  /**
+   * Event that triggers validation
+   */
+  validationTrigger?: 'input' | 'change' | 'focus' | 'blur';
+  /**
+   * Whether the input should be validated on mount
+   */
+  validateOnMount?: boolean;
+  /**
+   * Message to use as hint when validation fails
+   */
+  errorMessage?: string;
+  /**
+   * Loading state of the input
+   */
+  loading?: boolean;
+}
+
+const props = withDefaults(defineProps<FFileUploadProps>(), {
+  modelValue: null,
+  multiple: false,
+  iconName: 'upload',
+  size: 'medium',
+  buttonText: '',
+  allowedMimeTypes: () => [],
+  maximumSize: 50000,
+  rules: () => [],
+  color: 'neutral--light-3',
+  textColor: 'neutral--dark-4',
+  label: '',
+  labelTextColor: 'neutral--dark-4',
+  errorColor: 'danger',
+  hint: '',
+  hideHint: false,
+  hintIcon: null,
+  hintTextColor: 'neutral--dark-4',
+  disabled: false,
+  name: 'defaultName',
+  validateOnMount: false,
+  validationTrigger: 'change',
+  errorMessage: '',
+});
+
+const underlyingFileInputRef = ref<HTMLInputElement>();
+const fileUploadButtonRef = ref<InstanceType<typeof FButton>>();
+
+const { isValid, hint, value, handleValidation } = useFieldWithValidation(
+  props,
+  {
+    validateOnMount: props?.validateOnMount,
+    rules: [isValidFile],
+  }
+);
+
+const emit = defineEmits<{
+  (name: 'update:modelValue', value: File[] | null | undefined): void;
+  (name: 'change', value: Event): void;
+  (name: 'focus'): void;
+  (name: 'files', value: File[]): void;
+}>();
+
+function isValidFile(value: unknown): boolean {
+  const isValidFormat = mimes(value, props.allowedMimeTypes);
+  const isValidSize = sizeRule(value, { size: props.maximumSize });
+
+  const validFile = isValidFormat && isValidSize;
+  if (validFile) {
+    emit('files', value as File[]);
+  }
+
+  return validFile;
```

Comment:
If you use `is` before booleans, for consistency you should keep it. As you are in the scope of the funtion, the local `validFile` can be name `isValid` what do you think ?
```suggestion
  const isValid = isValidFormat && isValidSize;
  if (isValid) emit('files', value as File[]);
  return isValid;
```

Reply:
Even if it works, isn't it slightly misleading to have the same variable name in a file ?
I was thinking `isValidFormatAndSize` even though its a bit long and needs to be updated the day we add a 3rd rule .. 🤔

Reply:
Good point, go for `isValidFormatAndSize` 👍